src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -215,6 +217,12 @@ public void ShutdownSend()
         // Only call this after sending a GOAWAY.
         public async Task WaitForConnectionShutdownAsync()
         {
+            if (_shutdownDone)
+            {
+                return;
```

Comment:
do we expect this to be called multiple times? In most of the cases this is presumably a test issue. Perhaps makes sense to track this in the test which needs this directly

Reply:
it is also called from Dispose(). I may update Dispose() to call your new function to ignore errors but allowing it called multiple times may simplify cases with errors.

Reply:
I agree with @krwq here -- if you're calling this twice, it's probably a test issue, so it's better not to allow it to be called twice.

Reply:
I updated Dispose() to call newly added ShutdownIgnoringErrorsAsync() and removed the check.